src/parser.cpp

```diff
@@ -314,15 +314,16 @@ void Parser::ParserImpl::loadModel(const ModelPtr &model, const std::string &inp
     std::vector<XmlNodePtr> encapsulationNodes;
     while (childNode) {
         if (childNode->isCellmlElement("component")) {
-            ComponentPtr component = std::make_shared<Component>();
+            const std::string name;
+            ComponentPtr component = libcellml::Component::create(name);
             loadComponent(component, childNode);
             model->addComponent(component);
         } else if (childNode->isCellmlElement("units")) {
-            UnitsPtr units = std::make_shared<Units>();
+            UnitsPtr units = libcellml::Units::create();
```

Comment:
We shouldn't need to specify the `libcellml` namespace. IOW, I would expect to see:

```
UnitsPtr units = Units::create();
```

Same is true elsewhere in this PR.